src/alchemlyb/parsing/gomc.py

```diff
@@ -0,0 +1,211 @@
+    filename : str
+        Path to free energy file to extract data from.
+    T : float
+        Temperature in Kelvin at which the simulation was sampled.
+
+    Returns
+    -------
+    dH/dl : Series
+        dH/dl as a function of step for this lambda window.
+
+    """
+    beta = 1/(k_b * T)
+
+    state, lambdas, statevec = _extract_state(filename)
+
+    # extract a DataFrame from free energy data
+    df = _extract_dataframe(filename)
+
+    times = df[df.columns[0]]
+
+    # want to grab only dH/dl columns
+    dHcols = []
+    for l in lambdas:
+        dHcols.extend([col for col in df.columns if (l in col)])
+
+    dHdl = df[dHcols]
+
+    # make dimensionless
+    dHdl *= beta
+
+
+    dHdl = pd.DataFrame(dHdl.values, columns=lambdas,
+                        index=pd.Float64Index(times.values, name='time'))
+
+    # Need to modify the lambda name
+    cols = [l + "-lambda" for l in lambdas]
+    # create columns for each lambda, indicating state each row sampled from
+    for i, l in enumerate(cols):
+        try:
+            dHdl[l] = statevec[i]
+        except TypeError:
+            dHdl[l] = statevec
+
+    # set up new multi-index
+    newind = ['time'] + cols
+    dHdl= dHdl.reset_index().set_index(newind)
+
+    dHdl.name='dH/dl'
+
+    return dHdl
+
+
+def _extract_state(filename):
+    """Extract information on state sampled, names of lambdas.
+
+    """
+    state = None
+    with anyopen(filename, 'r') as f:
+        for line in f:
+            if ('#' in line) and ('State' in line):
+                state = int(line.split('State')[1].split(':')[0])
+                # GOMC always print these two fields
+                lambdas = ['Coulomb', 'VDW']
+                statevec = eval(line.strip().split(' = ')[-1])
+                break
+
+    return state, lambdas, statevec
+
+
+def _extract_dataframe(filename):
+    """Extract a DataFrame from free energy data.
+
+    """
+    dh_col_match = "dU/dL"
+    h_col_match = "DelE"
+    pv_col_match = 'PV'
+    u_col_match = 'Total_En'
+
+    xaxis = "time"
+    with anyopen(filename, 'r') as f:
+        names = []
+        rows = []
+        for line in f:
+            line = line.strip()
+            if len(line) == 0:
+                continue
+
+            if line.startswith("#Steps"):
+                element = line.split()
+                for i in range(len(element)):
+                    if element[i].startswith(u_col_match):
+                        names.append(element[i])
+                    elif element[i].startswith(dh_col_match):
+                        names.append(element[i])
+                    elif element[i].startswith(h_col_match):
+                        names.append(element[i])
+                    elif element[i].startswith(pv_col_match):
+                        names.append(element[i])
+
+            # should catch non-numeric lines so we don't proceed in parsing
+            # here
+            if line.startswith('#'):
+                continue
```

Comment:
This line is not tested.

Can you add a datafile to your dataset that triggers this?

Reply:
Comments that start with "should do this" should not be in well-tested code....

Reply:
These comments were carry out from gmx parser

Reply:
If you don't use it then remove it.

Reply:
Btw, fair enough, they should also not be in the gmx code ;-)

Reply:
:D. I use it actually. The first line of free energy data would trigger it.